### PR TITLE
Added a reverse option to overplot_*

### PR DESCRIPTION
**Pull request by NeutralKaon**
This is to permit reverse with `overplot` functions to work as per #155

### CODE REVIEW

**Comment by dfm** [comment]
Awesome - thanks! Can you add a test for this in `test_corner.py`? If it's not clear how to fit it in there, I'm happy to do it!

**Comment by NeutralKaon** [comment]
Hi @dfm  -- Thank you! I _think_ I've done this sensibly (and hopefully the but it's my first time with the CI / testing framework, so a few quick questions: 

1. I added my generated files [attached!] to `tests/baseline_images/test_corner/` but it looks like the CI server is failing because those I generated are at a different resolution to what they should be. For my future education, what's the right way to generate these in the first place? It is failing at the moment -- but should be a quick fix -- just think it's quicker for you to tell me how to do it! Sorry...

2. Secondly, the existing `test_reverse` function did not actually have `reverse=True` in it, but was just for reversing the axes range. I hope you don't mind that I added another "extended overplotting" test which is the basis of the reversed one. 

Thanks! 
-- J

**Comment by dfm** [comment]
This looks good! These plotting tests can be pretty flaky, but I think I know how to get those tests passing. I'll push a tweak to your branch shortly.

One other thing that we should probably add to this PR: I *think* we'll want to pass reverse here:

https://github.com/dfm/corner.py/blob/6dcfb7fca264c53f738dcbb20873c9857f487d98/src/corner/core.py#L377-L383

and here:

https://github.com/dfm/corner.py/blob/6dcfb7fca264c53f738dcbb20873c9857f487d98/src/corner/arviz_corner.py#L178

Anywhere else?

Do you want to add that and take a shot at adding a test for that too (passing both `truths` and `reverse` to `corner`)?

**Comment by NeutralKaon** [comment]
@dfm Done! Still failing because I have been a bad boy and generated the example images with what my (3.8) version of `jupyter notebook` creates rather than the CI server…_mea culpa_.

It looks good though; you're right, passing it in to truths inherently sensible. I don't see any other references to those functions elsewhere -- think that's it?

File: tests/test_corner.py [hunk 1]
```diff
@@ -157,6 +157,20 @@ def test_truths():
...
+    baseline_images=["reverse_truths"], remove_text=True, extensions=["png"]
+)
+def test_reverse_truths():
+    _run_corner(truths=[0.0, None, 0.15], reverse=True)
+
+
+@image_comparison(
+    baseline_images=["truths"], remove_text=True, extensions=["png"]
+)
+def test_truths():
```
Comment (dfm) [inline tests/test_corner.py:170]: It looks like this test is a duplicate of a previous one?

File: src/corner/core.py [hunk 2]
```diff
@@ -739,9 +759,15 @@ def overplot_points(fig, xs, **kwargs):
     xs = _parse_input(xs)
     K = len(xs)
     axes = _get_fig_axes(fig, K)
-    for k1 in range(K):
-        for k2 in range(k1 + 1, K):
-            axes[k2, k1].plot(xs[k1], xs[k2], **kwargs)
+    if reverse:
```
Comment (dfm) [inline src/corner/core.py:762]: I think there's an indexing issue here. We don't want the points to appear on the diagonal (on the histogram plots).

**Review by dfm — CHANGES_REQUESTED** [review]

File: tests/test_corner.py (hunk 1 above)
Comment (NeutralKaon) [reply tests/test_corner.py:170]: My bad -- it's been duplicated; I am really not entirely sure how -- just delete it

Comment (dfm) [reply tests/test_corner.py:170]: :+1: Will do!

File: src/corner/core.py (hunk 2 above)
Comment (NeutralKaon) [reply src/corner/core.py:762]: Again, my bad. This should be up to `k1` on line 764. I'll push again.

**Comment by dfm** [comment]
Thanks @NeutralKaon!
